engine/crates/federated-graph/src/federated_graph/v2.rs

```diff
@@ -0,0 +1,517 @@
+
+    /// All [input value definitions](http://spec.graphql.org/October2021/#InputValueDefinition) in the federated graph. Concretely, these are arguments of output fields, and input object fields.
+    pub input_value_definitions: Vec<InputValueDefinition>,
+
+    /// All the strings in the federated graph, deduplicated.
+    pub strings: Vec<String>,
+
+    /// All the field types in the federated graph, deduplicated.
+    pub field_types: Vec<FieldType>,
+
+    /// All composed directive instances (not definitions) in a federated graph.
+    pub directives: Vec<Directive>,
+}
+
+#[derive(serde::Serialize, serde::Deserialize, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct InputValueDefinition {
+    pub name: StringId,
+    pub field_type_id: FieldTypeId,
+    pub directives: Directives,
+    pub description: Option<StringId>,
+}
+
+#[derive(serde::Serialize, serde::Deserialize, Clone, PartialEq, PartialOrd)]
+pub enum Value {
+    String(StringId),
+    Int(i64),
+    Float(f64),
+    Boolean(bool),
+    EnumValue(StringId),
+    Object(Box<[(StringId, Value)]>),
+    List(Box<[Value]>),
+}
+
+#[derive(serde::Serialize, serde::Deserialize, Clone, PartialEq, PartialOrd)]
+pub struct Directive {
+    pub name: StringId,
+    pub arguments: Vec<(StringId, Value)>,
+}
+
+#[derive(serde::Serialize, serde::Deserialize, Clone)]
+pub struct Enum {
+    pub name: StringId,
+    pub values: Vec<EnumValue>,
+
+    /// All directives that made it through composition. Notably includes `@tag`.
+    pub composed_directives: Directives,
+
+    #[serde(default)]
+    pub description: Option<StringId>,
+}
+
+#[derive(serde::Serialize, serde::Deserialize, Clone)]
+pub struct EnumValue {
+    pub value: StringId,
+
+    /// All directives that made it through composition. Notably includes `@tag`.
+    pub composed_directives: Directives,
+
+    #[serde(default)]
+    pub description: Option<StringId>,
+}
+
+#[derive(serde::Serialize, serde::Deserialize, Clone)]
+pub struct Field {
+    pub name: StringId,
+    pub field_type_id: FieldTypeId,
+
+    pub arguments: InputValueDefinitions,
+
+    /// This is populated only of fields of entities. The Vec includes all subgraphs the field can
+    /// be resolved in. For a regular field of an entity, it will be one subgraph, the subgraph
+    /// where the entity field is defined. For a shareable field in an entity, this contains the
+    /// subgraphs where the shareable field is defined on the entity. It may not be all the
+    /// subgraphs.
+    ///
+    /// On fields of value types and input types, this is empty.
+    pub resolvable_in: Vec<SubgraphId>,
+
+    /// See [FieldProvides].
+    pub provides: Vec<FieldProvides>,
+
+    /// See [FieldRequires]
+    pub requires: Vec<FieldRequires>,
+
+    /// See [Override].
+    pub overrides: Vec<Override>,
+
+    /// All directives that made it through composition. Notably includes `@tag`.
+    pub composed_directives: Directives,
+
+    #[serde(default)]
```

Comment:
if you keep `#[serde(default)]` in the v2, you might want to skip to use `#[serde(default, skip_serializing_if = "Option::is_none")]`

Reply:
done in next commit